### DISCUSSION TITLE
missing usocket module

### DISCUSSION THREAD

**Discussion by JJavs5** [micropython/micropython#10274] [RP2040 / Pico]
Hello, I want to get http requests on RPI pico, but im missing usocket module, Ive read it should be available by default in the newest firmware so i downloaded it, but i still cant see it - is it possible to do that in RP2040? here is list of my modules with the most recent firmware.

![image](https://user-images.githubusercontent.com/53041849/208460606-c2b222ba-9526-45e3-b418-88f99f4abad0.png)

**Comment by robert-hh** [comment]
The RPi pico has not network capabilities. The PICO W has.  So wich board do you use?

**Reply by JJavs5** [reply]
the most popular one, just rp2040 i guess. I thought its possible to connect to network via USB?

**Reply by jimmo** [reply]
This is theoretically possible, but not supported (nor are there plans to support this).

The best way to achieve something like this right now is to use (USB) serial to talk to the host PC, and write some sort of program on the PC that reads/writes commands over serial.

**Comment by MrTinkerman** [comment]
I have been playing around with a Pico that has an Ethernet connector:
https://www.wiznet.io/product-item/w5100s-evb-pico/

With firmware:
https://github.com/Wiznet/RP2040-HAT-MicroPython/releases/tag/v1.0.5

I control attached hardware via a TCP socket

**Comment by klint-k** [comment]
MicroPython v1.19.1-896-g2e4dda3c2 on 2023-02-25; Raspberry Pi Pico W with RP2040
It also has a bad usocket!

  File "/lib/socket.py", line 47, in accept
AttributeError: 'module' object has no attribute 'sockaddr'


```
43  class socket(_socket.socket):
44  
45      def accept(self):
46          s, addr = super().accept()
**47          addr = _socket.sockaddr(addr)**
48          return (s, (_socket.inet_ntop(addr[0], addr[1]), addr[2]))

```
